### PR TITLE
ci: update Unified CI configuration

### PR DESCRIPTION
**Pull request by galargh**

### CODE REVIEW

**Comment by galargh** [comment]
I'm going to merge it after verifying all the flags were applied correctly.

**Review by Jorropo — APPROVED** [review]
LGTM

File: .github/workflows/go-test.yml [hunk 1]
```diff
@@ -30,6 +30,16 @@ jobs:
           submodules: recursive
       - id: config
         uses: protocol/.github/.github/actions/read-config@master
+      - if: toJSON(fromJSON(steps.config.outputs.json).shuffle) != 'false'
+        run: |
+          echo "GOTESTFLAGS=-shuffle=on $GOTESTFLAGS"
+          echo "GO386FLAGS=-shuffle=on $GO386FLAGS"
+          echo "GORACEFLAGS=-shuffle=on $GORACEFLAGS"
```
Comment (galargh) [inline .github/workflows/go-test.yml]:
```suggestion
          echo "GOTESTFLAGS=-shuffle=on $GOTESTFLAGS" >> $GITHUB_ENV
          echo "GO386FLAGS=-shuffle=on $GO386FLAGS" >> $GITHUB_ENV
          echo "GORACEFLAGS=-shuffle=on $GORACEFLAGS" >> $GITHUB_ENV
```

File: .github/workflows/go-test.yml [hunk 2]
```diff
@@ -30,6 +30,16 @@ jobs:
...
+      - if: toJSON(fromJSON(steps.config.outputs.json).verbose) != 'false'
+        run: |
+          echo "GOTESTFLAGS=-v $GOTESTFLAGS"
+          echo "GO386FLAGS=-v $GO386FLAGS"
+          echo "GORACEFLAGS=-v $GORACEFLAGS"
```
Comment (galargh) [inline .github/workflows/go-test.yml]:
```suggestion
          echo "GOTESTFLAGS=-v $GOTESTFLAGS" >> $GITHUB_ENV
          echo "GO386FLAGS=-v $GO386FLAGS" >> $GITHUB_ENV
          echo "GORACEFLAGS=-v $GORACEFLAGS" >> $GITHUB_ENV
```

**Review by galargh — COMMENTED** [review]
Forgot to actually update the env 🤦 I'm going to cherry-pick this fix in uCI later.

**Pull request merged**